### ISSUE TITLE
Prepare failed: Login error: EOF

### ISSUE THREAD

**Issue by rocsunny** [denisenkom/go-mssqldb#590]
hi,
DataBase: MS Sql server 2008 r2
Code : from sample
Error: Prepare failed: Login error: EOF

what can I do?

**Comment by chengshengqiing** [comment]
I've met too. If you can solve this problem, can I have it? Thank you

**Comment by dravenk** [comment]
I used the same version MS Sql server 2008 r2 and caught the same errors.
```
Login error: unexpected EOF
```

**Comment by edram1916** [comment]
In your connection parameters you should try "encrypt=disable"

**Comment by dravenk** [comment]
Yeah. I tried both "encrypt=disable" and "encrypt=false" in my connection string.
```
server=MYPC;user id=SA;password=MyPassword;port=1433;connection timeout=15;encrypt=disable;TrustServerCertificate=false;app name=myapp;log=32;
```
But unfortunately, it didn't work as expected. That's probably where the error came from.  
https://github.com/denisenkom/go-mssqldb/blob/master/tds.go#L1207
```
			case doneStruct:GameXG, 4 years ago: • fix #218
				if token.isError() {
					return nil, fmt.Errorf("login error: %s", token.getError())denisenk, 2 months ago: • fixed staticcheck errors
				}GameXG, 4 years ago: • fix #218
			case error:Rose, William, 29 days ago: • Add support for AD federated authentication l…
				return nil, fmt.Errorf("login error: %s", token.Error())
````
Maybe there should have been a more detailed error message than `unexpected EOF`

**Comment by edram1916** [comment]
I was having the same error 

> Login error: unexpected EOF



Then I declared my parameters like var

> dbServer   = "MyIP"
> dbPort     = 1433
> dbUser     = "User"
> dbPassword = "Pass"
> dbName     = "MyDB" 
>  

and my connection string:

> connString := fmt.Sprintf("server=%s;user id=%s;password=%s;port=%d;database=%s;encrypt=disable;", dbServer, dbUser, dbPassword, dbPort, dbName)

**Comment by dravenk** [comment]
It is possible that the error is not really from a login error. Because when I check the MSSQL log, I find a lot of records about the program that has been logged in.
```
Login succeeded for user "XX". Connection: Non-Trusted.
```

**Comment by dravenk** [comment]
Finally, I changed the connection string to ODBC, and I was able to connect to the database and execute the `SELECT @@version` statement.
```
  odbc:server=MYPC\Instance;user id=sa;password=saPassword;database=master;app name=myApp;
````

**Comment by KervinChang** [comment]
> Finally, I changed the connection string to ODBC, and I was able to connect to the database and execute the `SELECT @@version` statement.
> 
> ```
>   odbc:server=MYPC\Instance;user id=sa;password=saPassword;database=master;app name=myApp;
> ```

I encountered the same problem as you, but I got a prompt as you did -- "sql: database is closed"
